src/github.com/matrix-org/dendrite/federationsender/types/types.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
+// A JoinedHost is a server that is joined to a matrix room.
+type JoinedHost struct {
+	// THe EventID of a m.room.member event that joins a server to a room.
```

Comment:
Oh interesting. The event ID has to reference the `join` membership of a `m.room.member` event. This was not at all clear and makes your comments in another file make a bit more sense. I would be tempted to `s/EventID/MembershipEventID/` for clarity, at the cost of verbosity.